code/game/objects/items/ego_weapons/teth.dm

```diff
@@ -312,3 +312,100 @@
 	icon = 'icons/effects/effects.dmi'
 	icon_state = "smash1"
 	duration = 3
+
+#define LANTERN_MODE_REMOTE 1
+#define LANTERN_MODE_AUTO 2
+
+/obj/item/ego_weapon/lantern //meat lantern
```

Comment:
It reportedly glows in game, give it the light component or set the light variables so it does that please.